anomalib/models/cflow/lightning_model.py

```diff
@@ -159,3 +166,54 @@ def validation_step(self, batch, _):  # pylint: disable=arguments-differ
         batch["anomaly_maps"] = self.model(batch["image"])
 
         return batch
+
+
+class CflowLightning(Cflow):
+    """PL Lightning Module for the CFLOW algorithm.
+
+    Args:
+        hparams (Union[DictConfig, ListConfig]): Model params
+    """
+
+    def __init__(self, hparams: Union[DictConfig, ListConfig]) -> None:
+        super().__init__(
+            adaptive_threshold=hparams.model.threshold.adaptive,
+            default_image_threshold=hparams.model.threshold.image_default,
+            default_pixel_threshold=hparams.model.threshold.pixel_default,
+            input_size=hparams.model.input_size,
+            backbone=hparams.model.backbone,
+            layers=hparams.model.layers,
+            fiber_batch_size=hparams.dataset.fiber_batch_size,
+            decoder=hparams.model.decoder,
+            condition_vector=hparams.model.condition_vector,
+            coupling_blocks=hparams.model.coupling_blocks,
+            clamp_alpha=hparams.model.clamp_alpha,
+            permute_soft=hparams.model.soft_permutation,
+        )
+        self.hparams: Union[DictConfig, ListConfig]  # type: ignore
+        self.save_hyperparameters(hparams)
+
+    def configure_callbacks(self):
+        """Configure model-specific callbacks."""
+        early_stopping = EarlyStopping(
+            monitor=self.hparams.model.early_stopping.metric,
+            patience=self.hparams.model.early_stopping.patience,
+            mode=self.hparams.model.early_stopping.mode,
+        )
+        return [early_stopping]
+
+    def configure_optimizers(self) -> torch.optim.Optimizer:
+        """Configures optimizers for each decoder.
+
+        Returns:
+            Optimizer: Adam optimizer for each decoder
+        """
+        decoders_parameters = []
+        for decoder_idx in range(len(self.model.pool_layers)):
+            decoders_parameters.extend(list(self.model.decoders[decoder_idx].parameters()))
+
+        optimizer = optim.Adam(
+            params=decoders_parameters,
+            lr=self.hparams.model.lr,
+        )
+        return optimizer
```

Comment:
Since the idea is to deprecate this class with hparams in the future, I think the member functions should be in the base class. That way we can just delete the init function.

Reply:
I agree. There seems to be some inconsistency as well. Some classes have the methods in the base class, others in the Lightning class, yet others in a mixture of both.

Reply:
Ah, that explains it then. Thanks for clarifying :)

Maybe you could add this info to the docstring of the to-be-deprecated lightning classes to avoid confusion?